pyingest/parsers/oup.py

```diff
@@ -0,0 +1,250 @@
+#!/usr/bin/env python
+
+import sys
+import os
+import json
+import codecs
+import string
+from adsputils import u2asc
+from jats import JATSParser
+from pyingest.config.config import *
+from pyingest.parsers.entity_convert import EntityConverter
+from author_names import AuthorNames
+
+class NoSchemaException(Exception):
+    pass
+
+class WrongSchemaException(Exception):
+    pass
+
+class UnparseableException(Exception):
+    pass
+
+
+class OUPJATSParser(JATSParser):
+
+    AST_WORDS = [x.lower() for x in UAT_ASTRO_KEYWORDS]
+    AST_WORDS = AST_WORDS + [x.lower() for x in AAS_ASTRO_KEYWORDS]
+    AST_WORDS = AST_WORDS + [x.lower() for x in APS_ASTRO_KEYWORDS]
+
+
+    def get_author_init(self,namestring):
+        output = u2asc(namestring)
+        for c in output:
+            if c.isalpha():
+                return c.upper()
+        return u'.'
+
+    def oup_journals(self, pid):
+        try:
+            bibstem = OUP_PUBLISHER_IDS[pid]
+        except KeyError:
+            return 'XSTEM'
+        else:
+            return bibstem
+
+    def get_tmp_page(self, bs):
```

Comment:
I'm confused here.  I'm not sure if these tasks need to be done here at all -- are they needed for parsing or for updating index files?  If you need these to get parsing data, I'd recommend making a dict in config/config.py like OUP_LINKS_DIRS = {'mnrasl': '/proj/ads/abstracts/config/links//DOI/MNRASL', etc...}

Reply:
Also, a second issue is that the testing service built in to Github is going to fail because it doesn't have access to /proj

Reply:
OK, I'll move them into config - they are needed to get the "page" number for the tmp bibcodes, which is a sequential counter based on what is already in those files.

Reply:
The testing is going to fail anyway because it won't be able to create the same id number every time.  Can you suggest a better way to do it?